@xen-orchestra/backups/disks/RemoteVhd.mjs

```diff
@@ -141,6 +141,75 @@ export class RemoteVhd extends RandomAccessDisk {
       data,
     }
   }
+
+  /**
+   * Writes a full block into this VHD.
+   * @param {number} index
+   * @param {Buffer} data
+   */
+  async writeBlock(index, data) {
+    if (this.#vhd === undefined) {
+      throw new Error(`can't call readBlock of a RemoteVhd before init`)
+    }
+    await this.#vhd.writeEntireBlock({ id: index, buffer: data })
+    await this.#vhd.writeBlockAllocationTable()
```

Comment:
I think writing the block allocation table on each block write will be expensive ( it can be bigger than the block itself) and also pose some issue when writing blocks in parallel